### ISSUE TITLE
[bug] ConnectorNotFoundError triggered when query params are present or when navigating in Nextjs app

### ISSUE THREAD

**Issue by jjranalli** [tmm/wagmi#581]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.4.11

### Current Behavior

When opening a nextjs page with any query params in the url, or when navigating to other pages using Next/link, connector is `undefined` and signer is `null`.

This leads to bugs when trying to connect wallet (in my case RainbowKit popup getting stuck on "waiting for connection", or triggering `ConnectorAlreadyConnectedError`) as well as impossibility to sign messages via `useSignMessage`.

Same behaviour as #546, so the issues are probably related

### Expected Behavior

Navigating through pages or adding query params shouldn't influence connector or signer

### Steps To Reproduce

1. Go on a nextjs app either by adding a query param in the url or by navigating to it via Next/link
2. Check that return values of `useAccount` and `useSigner` are respectively `undefined` and `null`

### Anything else?

Using wagmi with RainbowKit, but issue seems unrelated.

**Comment by tmm** [comment]
If you check out https://github.com/tmm/wagmi/issues/546#issuecomment-1145200098, there's a solution to that issue. If that doesn't work, please create a minimal reproducible example and re-open this issue.

**Comment by jjranalli** [comment]
Yes that doesn't seem to work, also I'm calling `createClient` in the global scope.

Will provide a repo soon and reopen the issue, thanks.

**Comment by jjranalli** [comment]
Added repo to reproduce issue and deployment link @tmm

**Comment by tmm** [comment]
Doesn't look like `createClient` is called in the global scope [here](https://github.com/jjranalli/wagmi-connector-debug/blob/master/pages/_app.tsx#L14-L28=). Can you move it outside of `MyApp`?

**Comment by jjranalli** [comment]
oh 🤦‍♂️ that fixed it. Apologies for opening the issue and thanks for catching that!
